README.md

```diff
@@ -57,15 +57,85 @@ forum is a good place to meet the rest of the community.
 ## Bugs and Feature Requests
 
 - For public feedback, bug reports, and feature requests please file an issue
-  via [GitHub](https://github.com/inrupt/solid-client-access-grants-js/issues/).
+  via [GitHub](https://github.com/inrupt/solid-client-errors-js/issues/).
 - For non-public feedback or support inquiries please use the
   [Inrupt Service Desk](https://inrupt.atlassian.net/servicedesk).
 
+## Examples
+
+Integrators of this library will generally use the following pattern to throw
+a subclass of `ClientHttpError`, corresponding to the response status.
+
+```javascript
+import { handleErrorResponse } from "@inrupt/solid-client-errors";
+
+const response = await fetch(url, options);
+
+if (response.status !== 200) {
+  const responseBody = await response.text();
+  throw handleErrorResponse(
+    response,
+    responseBody,
+    "application error message",
+  );
+}
+```
+
+The specific error type corresponds to the HTTP status code in the response, each of
+which implements the `WithProblemDetails` type.
+
+- 400: `BadRequestError`
+- 401: `UnauthorizedError`
+- 403: `ForbiddenError`
+- 404: `NotFoundError`
+- 405: `MethodNotAllowedError`
+- 406: `NotAcceptableError`
+- 409: `ConflictError`
+- 410: `GoneError`
+- 412: `PreconditionFailedError`
+- 415: `UnsupportedMediaTypeError`
+- 429: `TooManyRequestsError`
+- 500: `InternalServerError`
+
+Each error includes a `problemDetails` field with the following properties:
+
+- `type: URL`: The specific error type. By default, this is `about:blank`.
+- `title: string`: A short description of the problem.
+- `status: number`: The error response status code.
+- `detail?: string`: A longer description of the problem.
+- `instance?: URL`: A unique URL identifying the problem occurrence.
+
+Consumers of these errors may handle the `ProblemDetails` data in the following way:
+
+```javascript
+const res = await getFile(url, options).catch((err) => {
+  const problems = err.problemDetails;
+  const header = document.createElement("h2");
+  header.appendChild(
+    document.createTextNode(`${problems.status} ${problems.title}`),
+  );
+
+  const el = document.getElementById("error");
+  el.appendChild(header);
+
+  if (problems.detail) {
+    const message = document.createElement("p");
+    message.appendChild(document.createTextNode(problems.detail));
+    el.appendChild(message);
+  }
+
+  if (problems.instance) {
+    const instance = document.createElement("p");
+    instance.appendChild(document.createTextNode(problem.instance));
+    el.appendChild(instance);
+  }
+});
```

Comment:
Here, I don't know if I would go into so much details about what to do with the Problem Details data, because that will vary vastly from one app to another. Instead, I would focus on the first couple of lines, something like
```ts
const res = await getFile(url, options)
  .then((file) => { /* Do something with the file after successful request */ })
  .catch((err) => {
    if (hasProblemDetails(err)) {
      const problems = err.problemDetails;
      /* Do something with the details of the failed request */
    } else {
      /* Do something with the generic error message */
    }
  });
```
This also showcases the type guard we want people to use.